File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/ColumnsShareDemo/ColumnsShareDemoMain.pas
unit ColumnsShareDemoMain;

{$I cxVer.inc}

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, cxControls, cxLookupGrid, cxLookupDBGrid, cxGridCustomTableView,
  cxGridTableView, cxGridDBTableView, cxGridLevel, cxGridCustomView, cxGrid,
  cxGridCardView, cxGridDBCardView, cxMaskEdit, StdCtrls, ExtCtrls,
  cxContainer, cxEdit, cxTextEdit, cxDropDownEdit, cxDBEdit, cxStyles,
  Menus, ActnList, ImgList, ComCtrls, cxLookAndFeels, DB,
  cxCustomData, cxGraphics, cxFilter, cxData, cxDBData, cxClasses,
  cxDBEditRepository, cxGridCustomPopupMenu, cxGridPopupMenu,
  cxDBExtLookupComboBox, cxDataStorage, cxImageComboBox,
  cxDBLookupComboBox, cxHyperLinkEdit, cxLookAndFeelPainters,
  cxGridCustomLayoutView, BaseForm;

type
  TColumnsShareDemoMainForm = class(TfmBaseForm)
    miOptions: TMenuItem;
    Grid: TcxGrid;
    tvProjects: TcxGridDBTableView;
    tvProjectsNAME: TcxGridDBColumn;
    tvProjectsMANAGERID: TcxGridDBColumn;
    tvItems: TcxGridDBTableView;
    tvItemsNAME: TcxGridDBColumn;
    tvItemsTYPE: TcxGridDBColumn;
    tvItemsPROJECTID: TcxGridDBColumn;
    tvItemsPRIORITY: TcxGridDBColumn;
    tvItemsSTATUS: TcxGridDBColumn;
    tvItemsCREATORID: TcxGridDBColumn;
    tvItemsCREATEDDATE: TcxGridDBColumn;
    tvItemsOWNERID: TcxGridDBColumn;
    tvItemsLASTMODIFIEDDATE: TcxGridDBColumn;
    tvItemsFIXEDDATE: TcxGridDBColumn;
    tvItemsDESCRIPTION: TcxGridDBColumn;
    cvUsers: TcxGridDBCardView;
    cvUsersFNAME: TcxGridDBCardViewRow;
    cvUsersMNAME: TcxGridDBCardViewRow;
    cvUsersLNAME: TcxGridDBCardViewRow;
    cvUsersCOUNTRY: TcxGridDBCardViewRow;
    cvUsersPOSTALCODE: TcxGridDBCardViewRow;
    cvUsersCITY: TcxGridDBCardViewRow;
    cvUsersADDRESS: TcxGridDBCardViewRow;
    cvUsersPHONE: TcxGridDBCardViewRow;
    cvUsersFAX: TcxGridDBCardViewRow;
    cvUsersEMAIL: TcxGridDBCardViewRow;
    cvUsersHOMEPAGE: TcxGridDBCardViewRow;
    tvTeam: TcxGridDBTableView;
    tvTeamUSERID: TcxGridDBColumn;
    tvTeamFUNCTION: TcxGridDBColumn;
    lvItems: TcxGridLevel;
    lvProjects: TcxGridLevel;
    lvUsers: TcxGridLevel;
    cxEditRepository: TcxEditRepository;
    eriLookupComboUsers: TcxEditRepositoryLookupComboBoxItem;
    lvTeam: TcxGridLevel;
    cxGridPopupMenu1: TcxGridPopupMenu;
    miCustomizePersonsLookupCombobox: TMenuItem;
    cvUsersDepartment: TcxGridDBCardViewRow;
    cxGridViewRepository: TcxGridViewRepository;
    cxGridViewRepositoryDBTableView: TcxGridDBTableView;
    cxGridViewRepositoryDBTableViewID: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewUserName: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewCOUNTRY: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewPOSTALCODE: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewCITY: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewADDRESS: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewPHONE: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewFAX: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewEMAIL: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewHOMEPAGE: TcxGridDBColumn;
    cxGridViewRepositoryDBTableViewDepartment: TcxGridDBColumn;
    erExtLookupComboBoxItem: TcxEditRepositoryExtLookupComboBoxItem;
    miPersonEditor: TMenuItem;
    miExtLookUpEditor: TMenuItem;
    miLookupEditor: TMenuItem;
    procedure FormShow(Sender: TObject);
    procedure miCustomizePersonsLookupComboboxClick(Sender: TObject);
    procedure miLookUpEditorClick(Sender: TObject);
    procedure miExtLookUpEditorClick(Sender: TObject);
  protected
    procedure LookAndFeelChanged(Sender: TcxLookAndFeel; AChangedValues: TcxLookAndFeelValues); override;
  end;

var
  ColumnsShareDemoMainForm: TColumnsShareDemoMainForm;

implementation

uses
  ColumnsShareDemoData, ColumnsShareDemoLookupCustomize, AboutDemoForm;

{$R *.dfm}

procedure TColumnsShareDemoMainForm.FormShow(Sender: TObject);
begin
  miCustomizePersonsLookupComboboxClick(nil);
end;

procedure TColumnsShareDemoMainForm.LookAndFeelChanged(
  Sender: TcxLookAndFeel; AChangedValues: TcxLookAndFeelValues);
begin
  inherited LookAndFeelChanged(Sender, AChangedValues);
  UpdateTableViewStyleSheet(tvProjects);
  UpdateTableViewStyleSheet(tvItems);
  UpdateTableViewStyleSheet(tvTeam);
  UpdateCardViewStyleSheet(cvUsers);
end;

procedure TColumnsShareDemoMainForm.miLookUpEditorClick(Sender: TObject);
begin
  if GetMenuItemChecked(Sender) then
  begin
    tvItemsCREATORID.RepositoryItem := cxEditRepository[1];
    tvItemsOWNERID.RepositoryItem := cxEditRepository[1];
    tvProjectsMANAGERID.RepositoryItem := cxEditRepository[1];
    tvTeamUSERID.RepositoryItem := cxEditRepository[1];
    tvProjects.DataController.ClearDetails;
    MenuItemSetEnabled('miCustomizePersonsLookupCombobox', True);
  end;
end;

procedure TColumnsShareDemoMainForm.miExtLookUpEditorClick(Sender: TObject);
begin
  if GetMenuItemChecked(Sender) then
  begin
    tvItemsCREATORID.RepositoryItem := cxEditRepository[0];
    tvItemsOWNERID.RepositoryItem := cxEditRepository[0];
    tvProjectsMANAGERID.RepositoryItem := cxEditRepository[0];
    tvTeamUSERID.RepositoryItem := cxEditRepository[0];
    tvProjects.DataController.ClearDetails;

    if (ColumnsShareDemoLookupCustomizeForm <> nil) and
      ColumnsShareDemoLookupCustomizeForm.Showing
    then
      ColumnsShareDemoLookupCustomizeForm.Hide;
    MenuItemSetEnabled('miCustomizePersonsLookupCombobox', False);
  end;
end;

procedure TColumnsShareDemoMainForm.miCustomizePersonsLookupComboboxClick(Sender: TObject);
begin
  if not Assigned(ColumnsShareDemoLookupCustomizeForm) then
    ColumnsShareDemoLookupCustomizeForm := TColumnsShareDemoLookupCustomizeForm.Create(Self);
  ColumnsShareDemoLookupCustomizeForm.Show;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/ColumnsShareDemo/ColumnsShareDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ColumnsShareDemoMain.h"
#include "ColumnsShareDemoData.h"
#include "AboutDemoForm.h"
#include "ColumnsShareDemoLookupCustomize.h"
#include "shellapi.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxCustomData"
#pragma link "cxData"
#pragma link "cxDBData"
#pragma link "cxEdit"
#pragma link "cxFilter"
#pragma link "cxGraphics"
#pragma link "cxGrid"
#pragma link "cxGridCustomTableView"
#pragma link "cxGridCustomView"
#pragma link "cxGridDBTableView"
#pragma link "cxGridLevel"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma link "cxGridCardView"
#pragma link "cxGridCustomPopupMenu"
#pragma link "cxGridDBCardView"
#pragma link "cxGridPopupMenu"
#pragma link "cxDBEditRepository"
#pragma link "cxDBExtLookupComboBox"
#pragma link "cxLookAndFeels"
#pragma link "cxDataStorage"
#pragma link "cxDBLookupComboBox"
#pragma link "cxGridCustomLayoutView"
#pragma link "cxHyperLinkEdit"
#pragma link "cxImageComboBox"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxMaskEdit"
#pragma resource "*.dfm"
TColumnsShareDemoMainForm *ColumnsShareDemoMainForm;
//---------------------------------------------------------------------------
__fastcall TColumnsShareDemoMainForm::TColumnsShareDemoMainForm(TComponent* Owner)
  : TfmBaseForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TColumnsShareDemoMainForm::FormShow(TObject *Sender)
{
  ColumnsShareDemoLookupCustomizeForm->Show();
}
//---------------------------------------------------------------------------

void __fastcall TColumnsShareDemoMainForm::miCustomizePersonsLookupComboboxClick(
	  TObject *Sender)
{
  ColumnsShareDemoLookupCustomizeForm->Show();
}
//---------------------------------------------------------------------------


void __fastcall TColumnsShareDemoMainForm::miLookUpEditorClick(
	  TObject *Sender)
{
  if (((TMenuItem*)Sender)->Checked) {
	tvItemsCREATORID->RepositoryItem = cxEditRepository->Items[1];
    tvItemsOWNERID->RepositoryItem = cxEditRepository->Items[1];
    tvProjectsMANAGERID->RepositoryItem = cxEditRepository->Items[1];
    tvTeamUSERID->RepositoryItem = cxEditRepository->Items[1];
	tvProjects->DataController->ClearDetails();
    miCustomizePersonsLookupCombobox->Enabled = true;
  }
}
//---------------------------------------------------------------------------

void __fastcall TColumnsShareDemoMainForm::miExtLookUpEditorClick(
      TObject *Sender)
{
  if (((TMenuItem*)Sender)->Checked) {
    tvItemsCREATORID->RepositoryItem = cxEditRepository->Items[0];
    tvItemsOWNERID->RepositoryItem = cxEditRepository->Items[0];
    tvProjectsMANAGERID->RepositoryItem = cxEditRepository->Items[0];
    tvTeamUSERID->RepositoryItem = cxEditRepository->Items[0];
	tvProjects->DataController->ClearDetails();
	if ((ColumnsShareDemoLookupCustomizeForm) &&
      ColumnsShareDemoLookupCustomizeForm->Showing)
      ColumnsShareDemoLookupCustomizeForm->Hide();
    miCustomizePersonsLookupCombobox->Enabled = false;
  }
}
//---------------------------------------------------------------------------

